src/main/java/uk/gov/hmcts/reform/prl/handlers/TransferToAnotherCourtEventHandler.java

```diff
@@ -0,0 +1,165 @@
+    private final SendgridService sendgridService;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CoreCaseDataApi coreCaseDataApi;
+
+    @EventListener(condition = "#event.typeOfEvent eq 'Transfer to another court'")
+    public void transferCourtEmail(final TransferToAnotherCourtEvent event) {
+        CaseData caseData = event.getCaseData();
+        if (caseData.getCourtEmailAddress() != null) {
+            sendTransferCourtEmail(caseData);
+            sendTransferToAnotherCourtEmail(event.getAuthorisation(),caseData);
+        }
+    }
+
+    private void sendTransferToAnotherCourtEmail(String authorization,CaseData caseData) {
+        try {
+            sendgridService.sendTransferCourtEmailWithAttachments(authorization,
+                                                                  getEmailProps(caseData),
+                                                                  caseData.getCourtEmailAddress(), getAllCaseDocuments(authorization,caseData));
+        } catch (IOException e) {
+            log.error("Failed to send Email");
+        }
+    }
+
+    private void sendTransferCourtEmail(CaseData caseData) {
+        emailService.send(
+            caseData.getCourtEmailAddress(),
+            EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION,
+            transferCaseContentProvider.buildCourtTransferEmail(caseData,getConfidentialityText(caseData)),
+            LanguagePreference.english
+        );
+    }
+
+    private Map<String, String> getEmailProps(CaseData caseData) {
+        Map<String, String> combinedMap = new HashMap<>();
+        combinedMap.put("caseNumber", String.valueOf(caseData.getId()));
+        combinedMap.put("caseName", caseData.getApplicantCaseName());
+        combinedMap.put("issueDate", CommonUtils.formatDate(D_MMMM_YYYY, caseData.getIssueDate()));
+        combinedMap.put("applicationType", caseData.getCaseTypeOfApplication());
+        combinedMap.put("confidentialityText", getConfidentialityText(caseData));
+        combinedMap.put("courtName", caseData.getTransferredCourtFrom());
+        combinedMap.putAll(getCommonEmailProps());
+        return combinedMap;
+    }
+
+    private String getConfidentialityText(CaseData caseData) {
+        return isCaseContainConfidentialDetails(caseData)
+            ? "This case contains confidential contact details." : "";
+    }
+
+    private boolean isCaseContainConfidentialDetails(CaseData caseData) {
+        return CollectionUtils.isNotEmpty(caseData.getApplicantsConfidentialDetails())
+            || CollectionUtils.isNotEmpty(caseData.getRespondentConfidentialDetails())
+            || CollectionUtils.isNotEmpty(caseData.getChildrenConfidentialDetails());
+    }
+
+    private Map<String, String> getCommonEmailProps() {
+        Map<String, String> emailProps = new HashMap<>();
+        emailProps.put("subject", "A case has been transferred to your court");
```

Comment:
move to constants?

Reply:
@shashisk97  as per your comment moved to constants